### PR TITLE
audit: Port line_problems to rubocop and add tests part 2

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----
#569 and continuation of #2964

### CODE REVIEW

File: Library/Homebrew/rubocops/lines_cop.rb [hunk 1]
```diff
@@ -15,6 +15,66 @@ def audit_formula(_node, _class_node, _parent_class_node, _body_node)
...
+          problem "Use a space in class inheritance: class #{@formula_name} < #{class_name(parent_class_node)}"
+        end
+      end
+
+      class Comments < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, _body_node)
+          # Commented-out cmake support from default template
+          audit_comments do |comment|
+            next unless comment.include?('# system "cmake')
+            problem "Commented cmake call found"
```
Comment (JCount) [inline Library/Homebrew/rubocops/lines_cop.rb]: Is there a reason to still keep this separate from the main `audit_comments` method for comments from the default template? It seems like they could probably be combined. @MikeMcQuaid any opinion here?

Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/lines_cop.rb]: Yeh, I'm fine either way but combining makes sense.

File: Library/Homebrew/rubocops/lines_cop.rb [hunk 2]
```diff
@@ -15,6 +15,66 @@ def audit_formula(_node, _class_node, _parent_class_node, _body_node)
...
+              next unless comment.include?(template_comment)
+              problem "Please remove default template comments"
+              break
+            end
+          end
+
+          audit_comments do |comment|
+            # Commented-out depends_on
+            next unless comment =~ /#\s*depends_on\s+(.+)\s*$/
+            problem "Commented-out dep #{Regexp.last_match(1)}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/rubocops/lines_cop.rb]: `dependency`

**Comment by MikeMcQuaid** [comment]
Great work @GauthamGoli! Will let @JCount 🚢 when he's happy.

**Comment by JCount** [comment]
👍 @GauthamGoli

**Pull request merged**